client/allocdir/alloc_dir.go

```diff
@@ -64,14 +91,20 @@ type AllocDirFS interface {
 	ChangeEvents(path string, curOffset int64, t *tomb.Tomb) (*watch.FileChanges, error)
 }
 
-func NewAllocDir(allocDir string) *AllocDir {
-	d := &AllocDir{AllocDir: allocDir, TaskDirs: make(map[string]string)}
+func NewAllocDir(allocDir string, maxSize int) *AllocDir {
+	d := &AllocDir{
+		AllocDir: allocDir,
+		maxSize:  maxSize,
+		TaskDirs: make(map[string]string),
+		watchCh:  make(chan struct{}),
```

Comment:
I've added them as exposed fields on the AllocDir struct. I also added a skeleton for a test but I'm not sure what you had in mind with the ModifyTime check. What would cause the ModifyTime to change?

Also, there is currently a test on AllocRunner that checks if the alloc is being destroyed when it exceeds disk space. Since the AllocDir creation is buried in the run code it is not possible to supply reduced times to the AllocDir before it starts. Any ideas to speed up the test?